src/components/CurrentUser/CurrentUser.tsx

```diff
@@ -1,12 +1,56 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './CurrentUser.scss';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
-export const CurrentUser: React.FC = () => (
-  <div className="CurrentUser">
-    <h2 className="CurrentUser__title"><span>Selected user: 2</span></h2>
+type Props = {
+  userId: number;
+  clearUser: () => void;
+};
 
-    <h3 className="CurrentUser__name">Ervin Howell</h3>
-    <p className="CurrentUser__email">Shanna@melissa.tv</p>
-    <p className="CurrentUser__phone">010-692-6593 x09125</p>
-  </div>
-);
+export const CurrentUser: React.FC<Props> = ({ userId, clearUser }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [errorMess, setErrorMess] = useState<string>('');
+
+  useEffect(() => {
+    getUser(userId)
+      .then((userFromServer) => setUser(userFromServer))
+      .catch((error) => {
+        setUser(null);
+        setErrorMess(`ERROR: Can't find user with id #${userId}!
+          <${error}>`);
+      });
+  }, [userId]);
+
+  return (
+    <>
+      {user
+        ? (
+          <div className="CurrentUser">
+            <h2 className="CurrentUser__title">
+              <span>{`Selected user: ${user.id}`}</span>
+            </h2>
+
+            <h3 className="CurrentUser__name" data-cy="userName">
+              {user.name}
+            </h3>
+            <p className="CurrentUser__email">{user.email}</p>
+            <p className="CurrentUser__phone">{user.phone}</p>
+
+            <button
+              type="button"
+              className="button CurrentUser__button-clear"
+              onClick={() => {
+                clearUser();
+                setUser(null);
+              }}
```

Comment:
move all inline functions to variable. Please check your code and improve according this comment